guides/payments/payment-processing.md

```diff
@@ -0,0 +1,167 @@
+# Payment processing
+
+Solidus does not process payments. Instead, it relies on [payment service
+providers][psp] like Stripe or Braintree. So, in order to process payments, your
+[payment methods][payment-methods] should first integrate with, and send
+payment data to, your payment service provider.
+
+While the `Spree::Payment` model executes much of the processing logic and
+manages the state of the payment (before and after processing), note that it
+includes the [`Spree::Payment::Processing` class][spree-payment-processing].
+
+<!-- TODO:
+  Add links to payment service providers article in this introduction once it is
+  merged.
+-->
+
+[payment-methods]: payment_method.md
+[payment-service-providers]: payment-service-providers.md
+[psp]: https://en.wikipedia.org/wiki/Payment_service_provider
+[spree-payment-processing]: https://github.com/solidusio/solidus/blob/master/core/app/models/spree/payment/processing.rb
+
+## Interfacing with a payment service provider
+
+When you set up your own payment methods, you need it to integrate with a payment
+service provider. While you can create an integration in any way that you want,
+you may want to note the following conventions set up by Solidus's
+`Spree::PaymentMethod` and `Spree::Payment::Processing` classes:
+
+- `Spree::PaymentMethod` has a similar interface to API provided by
+  [active_merchant][active-merchant].
+- The `Spree::PaymentMethod#gateway` method uses the
```

Comment:
Maybe I'm wrong, but seems like `gateway` method returns an instance of this payment method. Gateway_options is the variable provided to the gateway class.

Reply:
Thank you for making a note of this. While the `options` value is created by the `Spree::Payment::Processing#gateway_options` the way that I wrote this didn't really make sense.

I'm going to simplify it shortly and leave explanations to the "Gateway options" section below.